stablehlo/tests/interpret_probe.mlir

```diff
@@ -0,0 +1,84 @@
+// RUN: stablehlo-translate --interpret --probe-output-dir=%T -split-input-file %s
+
+func.func @probe_i1() {
+  %0 = stablehlo.constant dense<[[0], [0], [0]]> : tensor<3x1xi1>
+  %1 = stablehlo.constant dense<[[1], [1], [1]]> : tensor<3x1xi1>
+  %2 = stablehlo.add %0, %1 : tensor<3x1xi1>
+  %3 = interpreter.probe %2 { probe_id = "probe_i1" } : (tensor<3x1xi1>) -> tensor<3x1xi1>
```

Comment:
Optional QoL improvement:
Would be nice if we can write:
`%3 = interpreter.probe %2 { probe_id = "probe_i1" } : (tensor<3x1xi1>) -> tensor<3x1xi1>`
to something shorter:
`%3 = interpreter.probe %2, id = "probe_i1" : tensor<3x1xi1>`
or even:
`%3 = interpreter.probe %2, "probe_i1" : tensor<3x1xi1>`

I think this is just a 1 line change in the InterpreterOps.td file:
```
let assemblyFormat = "$operand `,` $probe_id attr-dict `:` type($operand)";
```